app/assets/javascripts/discourse/app/components/search-menu.hbs

```diff
@@ -4,62 +4,78 @@
-      @name="search-menu-before-term-input"
-      @outletArgs={{hash openSearchMenu=this.open}}
-    />
+      {{#if this.search.inTopicContext}}
+        <DButton
+          @icon="xmark"
+          @label="search.in_this_topic"
+          @title="search.in_this_topic_tooltip"
+          @action={{this.clearTopicContext}}
+          class="btn-small search-context"
+        />
+      {{else if this.inPMInboxContext}}
+        <DButton
+          @icon="xmark"
+          @label="search.in_messages"
+          @title="search.in_messages_tooltip"
+          @action={{this.clearPMInboxContext}}
+          class="btn-small search-context"
+        />
+      {{/if}}
 
-    <SearchMenu::SearchTerm
-      @searchTermChanged={{this.searchTermChanged}}
-      @typeFilter={{this.typeFilter}}
-      @updateTypeFilter={{this.updateTypeFilter}}
-      @triggerSearch={{this.triggerSearch}}
-      @fullSearch={{this.fullSearch}}
-      @clearPMInboxContext={{this.clearPMInboxContext}}
-      @clearTopicContext={{this.clearTopicContext}}
-      @closeSearchMenu={{this.close}}
-      @openSearchMenu={{this.open}}
-      @autofocus={{@autofocusInput}}
-    />
+      <PluginOutlet
+        @name="search-menu-before-term-input"
+        @outletArgs={{hash openSearchMenu=this.open}}
+      />
 
-    {{#if this.loading}}
-      <div class="searching">
-        {{loading-spinner}}
-      </div>
-    {{else}}
-      <div class="searching">
-        <PluginOutlet @name="search-menu-before-advanced-search" />
-        {{#if this.search.activeGlobalSearchTerm}}
-          <SearchMenu::ClearButton @clearSearch={{this.clearSearch}} />
-        {{/if}}
-        <SearchMenu::AdvancedButton
-          @openAdvancedSearch={{this.openAdvancedSearch}}
-        />
-      </div>
+      <SearchMenu::SearchTerm
+        @searchTermChanged={{this.searchTermChanged}}
+        @typeFilter={{this.typeFilter}}
+        @updateTypeFilter={{this.updateTypeFilter}}
+        @triggerSearch={{this.triggerSearch}}
+        @fullSearch={{this.fullSearch}}
+        @clearPMInboxContext={{this.clearPMInboxContext}}
+        @clearTopicContext={{this.clearTopicContext}}
+        @closeSearchMenu={{this.close}}
+        @openSearchMenu={{this.open}}
+        @autofocus={{@autofocusInput}}
+        data-test-input="search-term"
+      />
+
+      {{#if this.loading}}
+        <div class="searching">
+          {{loading-spinner}}
+        </div>
+      {{else}}
+        <div class="searching">
+          <PluginOutlet @name="search-menu-before-advanced-search" />
+          {{#if this.search.activeGlobalSearchTerm}}
+            <SearchMenu::ClearButton @clearSearch={{this.clearSearch}} />
+          {{/if}}
+          <SearchMenu::AdvancedButton
+            @openAdvancedSearch={{this.openAdvancedSearch}}
+          />
+        </div>
+      {{/if}}
+    </div>
+    {{#if this.site.isMobileViewAndDevice}}
+      <DButton
+        @action={{this.cancelSearchMobile}}
+        @label="cancel"
```

Comment:
Go for `cancel_value` here please, that's "Cancel" instead of "cancel"...we are gradually moving towards the guidelines here https://meta.discourse.org/t/formatting-text-in-discourse-documentation-and-uis/324637

Reply:
This is a random thought, but I do wonder how cramped this search will get in other languages with this Cancel button...you don't have to fix this necessarily, just something for future